src/lib/Hydra/Controller/Channel.pm

```diff
@@ -0,0 +1,83 @@
+package Hydra::Controller::Channel;
+
+use strict;
+use warnings;
+use base 'Hydra::Base::Controller::REST';
+
+
+sub channel : Chained('/') PathPart('channel/custom') CaptureArgs(3) {
+    my ($self, $c, $projectName, $jobsetName, $channelName) = @_;
+
+    $c->stash->{project} = $c->model('DB::Projects')->find($projectName);
+
+    notFound($c, "Project $projectName doesn't exist.")
+        if !$c->stash->{project};
+
+    $c->stash->{jobset} = $c->stash->{project}->jobsets->find({
+        name => $jobsetName
+    });
+
+    notFound($c, "Jobset $jobsetName doesn't exist.")
+        if !$c->stash->{jobset};
+
+    my $lastSuccessful = $c->model('DB::Builds')->find(
+        { 'eval.hasnewbuilds' => 1
+        , project => $projectName
+        , jobset => $jobsetName
+        , job => $channelName
+        , buildstatus => 0
+        },
+        { rows => 1, order_by => "eval.id desc"
+        , join => { jobsetevalmembers => 'eval' }
+        }
+    );
```

Comment:
Okay, looked into that in detail... my original idea was that a particular channel doesn't need to be part of a finished evaluation as long as all builds needed for that particular channel are finished. For my use-case, this is what I want it to be, but if you prefer only fully built evals, but if you want I could add `finished => 1` to the conditions.